[CLC-169] Add jet job tutorial: updates (#244)
hazelcast/hazelcast-commandline-client: docs/modules/ROOT/pages/jet-job-management.adoc

- JRE 8 or better.
- Gradle 8 or better.

[[step-1-dev-configure]]
== Step 1. Add Configuration for the Development Cluster

Connect to the cluster with the following command:

[tabs]
====
{hazelcast-cloud}::
. Follow xref:managing-viridian-clusters.adoc[] to see how you can create and/or list your {hazelcast-cloud} clusters.
. Import the connection configuration for your {hazelcast-cloud} cluster CLUSTER_NAME and save it as `dev`.
+
[source,shell]
----
clc viridian import-config $CLUSTER_NAME --name dev
----
--
Other Clusters::
+
--
[source,shell]
----
clc config add dev cluster.name=<CLUSTER_NAME> cluster.address=<CLUSTER_ADDRESS>
----
--
====

[[step-2-build-jet-job]]
== Step 2. Build the Jet Job

The sample Jet job we use in this tutorial generates a stream of numbers and creates a link:https://hashids.org/[Hashid] for each number. The number and hashid pair is then saved to a Map with they number as the key and hashid as the value.

You can find the Jet job project in https://github.com/hazelcast/hazelcast-commandline-client[the CLC repository] under the path `examples/platform/simple-streaming-pipeline`.

> 

Author: I'll merge this PR without this step, but but will add that after releasing v5.3.0

Build the project with the following command:

[source,shell]
----
gradle shadowJar
----

> 

Author: I'll merge this PR without this output, but but will add that after releasing v5.3.0

[[step-3-submit-jet-job]]
== Step 3. Submit the Jet job

Submit the Jet job with the following command. It will wait for the job to be in the `RUNNING` state.
 
[source,shell]
----
clc -c dev job submit build/libs/simple-streaming-pipeline-1.0-SNAPSHOT-all.jar --wait
----

Now, list the jobs in the cluster. You should see `simple-pipeline` in the `RUNNING` state

[source,shell]
----
clc -c dev job list
----

Observe that the size of the `my-map` map continuously changes.

[source,shell]
----
while true; do \
  clc -c dev map size -n my-map; sleep 3; \
done
----

[[step-4-cancel-jet-job]]
== Step 4. Cancel the Jet job

Now, cancel the Jet job.

[source,shell]
----
clc -c dev job cancel prime-job
----

> Observe that the size of `my-map` doesn't increase anymore.

Reviewer: ```suggestion
. See how the size of the `my-map` remains the same.
```

[source,shell]
----
while true; do \
  clc -c dev map size -n my-map; sleep 3; \
done
----

== Summary

In this tutorial, you learned how to do the following:

* Connect to a Hazelcast cluster.

> * Build and submit Hazelcast Jet pipelines.

Reviewer: ```suggestion
* Build and submit Hazelcast Jet jobs.
```

> * Manage Jet pipeline lifecycle with commands: list and cancel.

Reviewer: ```suggestion
* Manage the lifecycle of a Jet pipeline using list and cancel commands.
```
